controllers/limitador_controller.go

```diff
@@ -199,6 +200,16 @@ func (r *LimitadorReconciler) reconcileDeployment(ctx context.Context, limitador
 		return err
 	}
 
+	// TODO: To be deleted when the upgrade path is no longer needed.
+	newDeployment := &appsv1.Deployment{}
+	if err := r.Client().Get(ctx, client.ObjectKeyFromObject(deployment), newDeployment); err != nil {
```

Comment:
Just after creating the deployment, it is unlikely that the object does not exist. However, this is not enforced. k8s is not like transactional databases. You can expect to return not found error. Then you should requeue and let the controller try again.

Reply:
Are we happy to have it requeue by just returning an error as is done here? otherwise we would need to change the type signature of this function to be able to get context as to whether there's error, no error or we should requeue.